airflow/www/templates/airflow/dag.html

```diff
@@ -153,11 +153,11 @@ <h4 class="pull-right" style="user-select: none;-moz-user-select: auto;">
                   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}">
                   <input type="hidden" name="dag_id" value="{{ dag.dag_id }}">
                   <input type="hidden" name="unpause" value="True">
-                  <input type="hidden" name="origin" value="{{ url_for('Airflow.' + dag.default_view, dag_id=dag.dag_id) }}">
+                  <input type="hidden" name="origin" value="{{ request.full_path }}">
```

Comment:
Argh, full_path includes the  query string, which is a bit of a problem.

If you have changed the date controls and are viewing an "old" run (in time, not out of date) then the run you have just triggered won't be visible. Not great UX.

Reply:
Agreed, I'll rework it.